fundamentals/csharp/lab-2/README.md

```diff
@@ -0,0 +1,136 @@
+# Lab 2: Resources, Resource Providers, and Language Hosts
+
+Let's talk about resources, resource providers, and language hosts. Learn more on the [Learn pathway](https://www.pulumi.com/learn/pulumi-fundamentals/create-docker-images/) if you're walking through this workshop alone!
+
+## Verify your application
+
+The application we'll be running on our infrastructure is in the [pulumi/tutorial-pulumi-fundamentals repo](https://github.com/pulumi/tutorial-pulumi-fundamentals) in the `app/` directory. Examine the Dockerfiles in each directory.
+
+<details>
+<summary><b>Question:</b> What's the Dockerfile in the `backend` directory doing?</summary>
+
+<br/>
+<b>Answer:</b> This Dockerfile copies the REST backend into the Docker filesystem, installs the dependencies, and builds the image. Note that port 3000 must be open on your host machine.
+</details>
+<br/>

```

Comment:
This is neat. We should do more of this. :)

Reply:
thanks!